sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/Expression.scala

```diff
@@ -565,7 +565,7 @@ abstract class UnaryExpression extends Expression with UnaryLike[Expression] {
    * of evaluation process, we should override [[eval]].
    */
   protected def nullSafeEval(input: Any): Any =
-    throw QueryExecutionErrors.notOverrideExpectedMethodsError("UnaryExpressions",
+    throw QueryExecutionErrors.notOverrideExpectedMethodsError("UnaryExpression",
```

Comment:
How about to pass the actual class name instead of the parent one.

Reply:
Good suggestion, this can eliminate ambiguity that `UnaryExpression` can override `eval`, in fact, the actual class override it.